packages/ses/src/module-load.js

```diff
@@ -348,13 +348,13 @@ function asyncJobQueue() {
    *
    * @returns {Promise<Array<Error>>}
    */
-  const drainQueue = async () => {
+  async function drainQueue() {
```

Comment:
Why change from arrow function to `function` function? Please don't do this.

Reply:
ah to support Hermes, otherwise we run into the other `SyntaxError`
- [`async functions are unsupported`](https://github.com/facebook/hermes/blob/main/lib/IRGen/ESTreeIRGen-func.cpp#L213-L217)
  - https://github.com/facebook/hermes/issues/1395

Reply:
if better to preserve original behaviour was thinking similar approach could work

try an async arrow fn, use it if ok
otherwise catch the error, if it's the SyntaxError we're expecting, use the non-arrow async fn

Reply:
This is a good reason to CI against the build of Hermes needed in prod. Mark and I will at least always be stubbing our toe moving Chesterton’s Fence with our strong preference for the arrow function, breaking Hermes, if there’s no evidence we broke it in CI.

Reply:
fair enough ^ Hermes CI will only pass atm with 3 parts of SES disabled noted in the [tracker](https://github.com/endojs/endo/issues/1891) (assertDirectEvalAvailable, completePrototypes, whitelistIntrinsics), but worth adding now so we have the evidence there

Reply:
> if better to preserve original behaviour was thinking similar approach could work
> 
> try an async arrow fn, use it if ok
> otherwise catch the error, if it's the SyntaxError we're expecting, use the non-arrow async fn

RE this approach, i'll note relevant conditions here (for using non-arrow functions on Hermes)
- `Error.prototype.jsEngine === 'hermes'`
- `e.name === 'SyntaxError'`
- `e.message.includes('async functions are unsupported')`

Reply:
and just noting this also applies to the [`load`](https://github.com/endojs/endo/pull/2334#discussion_r1662303982) function, as well as `drainQueue`

Reply:
We decided at today’s meeting to continue using async arrow functions and patch them up in a hermes specific transform for distribution.

This also opens us up to the option of having alternate implementations of some features using `hermes` conditions in `package.json` `imports`.